crates/core/app/src/tests/spend.rs

```diff
@@ -60,6 +62,7 @@ async fn spend_happy_path() -> anyhow::Result<()> {
     spend.check_stateless(transaction_context).await?;
     spend.check_stateful(state.clone()).await?;
     let mut state_tx = state.try_begin_transaction().unwrap();
+    state_tx.put_mock_source(1u8);
```

Comment:
This has to be inserted now that the SCT logic tracks provenance, otherwise execution will fail, because in the mock setting, we're only running a single action in isolation, which wouldn't pick up the transaction-level source setting.